pkg/reader/ccip.go

```diff
@@ -30,6 +30,9 @@ import (
 // TODO: unit test the implementation when the actual contract reader and writer interfaces are finalized and mocks
 // can be generated.
 type ccipChainReader struct {
+	// to prevent copying this struct.
+	_ sync.Mutex
```

Comment:
Are you sure this works?

Reply:
Sounds like there are some linters that prevent the copying, so better than nothing.

Reply:
Yeah, its linter based, not build based unfortunately.